virtool/uploads/data.py

```diff
@@ -45,14 +48,36 @@ async def find(self, user, upload_type, ready) -> UploadSearchResult:
             if ready is not None:
                 filters.append(SQLUpload.ready == ready)
 
-            results = await session.execute(select(SQLUpload).filter(*filters))
+            skip_count = 0
+
+            if page > 1:
+                skip_count = (page - 1) * per_page
+
+            results = await session.execute(
+                select(SQLUpload).filter(*filters).offset(skip_count).limit(per_page)
+            )
+
+            total_count = await session.execute(select(SQLUpload))
+
+            found_count = await session.execute(select(SQLUpload).filter(*filters))
```

Comment:
The way you are getting the counts is not optimal. Please figure out how to just get the counts via query instead of loading all of the results and using `len()` in Python.

Can we do this with fewer queries? I feel it is likely we can do something similar to `aggregate` in MongoDB where we can get multiple values back from a single query. If we are stuck with multiple queries, using `asyncio.gather` to run them concurrently instead of  serially.